app/src/main/java/org/mozilla/focus/fragment/UrlInputFragment.java

```diff
@@ -182,8 +185,33 @@ public boolean onPreDraw() {
 
         urlView.setOnCommitListener(this);
 
+        // Should we show the search terms or a URL in the urlView?
         if (session != null) {
-            urlView.setText(session.getUrl().getValue());
+            final String searchURL = session.getSearchUrl();
+            final String currentURL = session.getUrl().getValue();
+            final String searchTerms = session.getSearchTerms();
+            if (!TextUtils.isEmpty(currentURL)) {
+                if (session.isSearch()) {
+                    if ((!TextUtils.isEmpty(searchTerms))) {
+                        urlView.setText(searchTerms);
+                        session.setSearch(false);
+                        session.setSearchUrl(currentURL);
```

Comment:
I set this to false so when the user navigates away from their search by clicking on another link, the search terms will not keep showing up. But if they just click away and click back on the UrlInputFragment, then they will show up because the URL matches the search URL.